### PR TITLE
fix(smart-contracts): reset original price when a key is cancelled

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This PR resets the `_originalPrices` in case of explicit expiration/cancellation (on top of `_originalDurations`), to prevent future key renewals

# Issues

Fixes #
Refs #8692

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: smart-contracts/contracts/mixins/MixinRefunds.sol [hunk 1]
```diff
@@ -108,6 +108,7 @@ contract MixinRefunds is
 
     // make future reccuring transactions impossible
     _originalDurations[_tokenId] = 0;
+    _originalPrices[_tokenId] = 0;
```
Comment (julien51) [inline smart-contracts/contracts/mixins/MixinRefunds.sol:111]: Oh, now that I think about it I wonder what happens on renewals for a lock that has free keys?

**Review by julien51 — APPROVED** [review]
Please, add a test :)

File: smart-contracts/contracts/mixins/MixinRefunds.sol (hunk 1 above)
Comment (clemsos) [reply smart-contracts/contracts/mixins/MixinRefunds.sol:111]: yes why I using with duration together is safer

**Comment by clemsos** [comment]
> Please, add a test :)

actually we are already testing that the tx reverts if there is changes in the lock. The revert code is `LOCK_HAS_CHANGED` and as we check that none of duration, price and token address have changed at once, it makes it impossible to dissociate the exact reason why it reverts in test.

what we could do will be add an error code like `LOCK_HAS_CHANGED(0)` for change in duration, `LOCK_HAS_CHANGED(1)` for a change in price, and `LOCK_HAS_CHANGED(2)` is the token address changed.

@julien51 wdyt?

https://github.com/unlock-protocol/unlock/blob/d62617027b94b25914a7bc1c3822c4318882ffc7/smart-contracts/test/Lock/renewMembershipFor.js#L277-L290

**Comment by julien51** [comment]
I don't think that's necessary and would add weight... so I think we're good!

**Pull request merged**